### PR TITLE
Add Xeno carrier

### LINKED ISSUE #252: Carrier
- [x] #287
- [x] #288
- [x] #289

### PR DESCRIPTION
**Pull request by blueDev2**
## About the PR

CM13 parity
Resolves #252
## Media

https://vimeo.com/1009700506?share=copy
## Requirements

- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: blueDev2
- add: Add Carriers
- add: Add Resin Holes

### CODE REVIEW

File: Content.Shared/_RMC14/Xenonids/Egg/EggRetriever/XenoEggRetrieverComponent.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+[RegisterComponent]
+public sealed partial class XenoEggRetrieverComponent : Component
+{
+    public const string EggContainerId = "eggs";
+
+    [DataField]
+    public EntProtoId EggPrototype = "XenoEgg";
+
+    [DataField]
+    public int MaxEggs = 7;
```
Comment (Vermidia) [inline Content.Shared/_RMC14/Xenonids/Egg/EggRetriever/XenoEggRetrieverComponent.cs:22]: Very minor but max eggs is 8 [can check caste code here](https://github.com/cmss13-devs/cmss13/blob/d226d88e7ddb38e1ad2169852f0f99682b44ee5a/code/modules/mob/living/carbon/xenomorph/castes/Carrier.dm#L101)

**Comment by Kadeo64** [comment]
real

**Comment by KingOnBudget** [comment]
sweep

File: Content.Shared/_RMC14/Xenonids/Egg/EggRetriever/XenoEggRetrieverComponent.cs (hunk 1 above)
Comment (blueDev2) [reply Content.Shared/_RMC14/Xenonids/Egg/EggRetriever/XenoEggRetrieverComponent.cs]: How out of date is this wiki? 
Thanks for telling me.

**Comment by DangerRevolution** [comment]
image

Traps are incredibly obvious, they're highlighted by a mouse over.

Carrier throw is *way* too far; CM13 only has it at four tiles when using "throw" ability. It's only one tile using an actual "throw"

**Comment by Vermidia** [comment]
I plan to work on the parasites dying part (and in general making their AI and mechanics even more inline with cm-13)

File: Resources/Locale/en-US/_RMC14/xeno/xeno-abilities.ftl [hunk 2]
```diff
@@ -114,6 +114,16 @@ cm-xeno-invisibility-expire = We feel our invisibility end!
 cm-xeno-ovipositor-attach = We start attaching to the ovipositor.
 cm-xeno-ovipositor-detach = We start detaching from the ovipositor.
 
+# Retrieve Egg
+cm-xeno-retrieve-egg-no-eggs = We don't have any eggs stored!
+cm-xeno-retrieve-egg-too-many-eggs = We don't have space to store any more eggs!
+cm-xeno-retrieve-egg-stash-egg = We put the egg into our internal storage.
```
Comment (Vermidia) [inline Resources/Locale/en-US/_RMC14/xeno/xeno-abilities.ftl:120]: Generally carrier messages when getting or retreiving eggs/parasites are like this:
"We grab one of the facehugger in our storage. Now sheltering: [huggers_cur] / [huggers_max]."
They show you how much space you have left

**Comment by deltanedas** [comment]
> Carrier throw is way too far; CM13 only has it at four tiles when using "throw" ability. It's only one tile using an actual "throw"

this applies to drones too, i think cranking parasite mass up would make it go less far

**Comment by Vermidia** [comment]
Throwing will be impossible for non-carriers non-abilities in my parasite PR (just to make carrier throw the only way and because messing with mass would effect both throws and other stuff)

**Comment by DangerRevolution** [comment]
> Throwing will be impossible for non-carriers non-abilities in my parasite PR (just to make carrier throw the only way and because messing with mass would effect both throws and other stuff)
> 
> Losing throw when they could only throw 1 tile anyway seems fine to me

if you wouldn't mind, eggs are also in this, they can be thrown only one tile (no matter the circumstance)

**Comment by Vermidia** [comment]
Also why is tailstab removed? They have it in cm-13
image

**Comment by DangerRevolution** [comment]
> Also why is tailstab removed? They have it in cm-13 image

Y'know I thought they didn't have it because I never look in the top left of the hotbar

**Comment by DangerRevolution** [comment]
# PR Issues

- Clicking on a resin hole with an empty hand makes a doafter and nothing else.
- Someone infected with a parasite is still knocked by parasite traps (they shouldn't trigger, same as eggs).
- You can pick up a dead parasite; but not drag it.
- Eggs & Traps can be placed under items not intended to have traps under them (AKA no parity in trap placement rules)
- Carrier needs some sort of visual indicator regarding huggers; and eventually eggs. Both for clarity for the Carrier player and other people around them.
- It is possible to destroy the weeds under a trap and just have the lone trap left behind.
- Destroying a weed node removes the weeds under a trap (the weeds should stay around the trap)
- Uprooting a weed node; when placing a trap under it, just removes the node and the weed and puts a trap over the naked tile
- There is no message in the Hive Chat for:
  "(Rank) Facehugger has hugged a host at X Location"
  "Facehugger trap was triggered at X Location"
  "Acid trap was triggered at X Location"
  "Gas Trap was triggered at X Location"
  "Facehugger trap was triggered at X Location"
   followed by: "Facehugger has hugged a host at X Location"
   "X trap was burnt at X location"
- Using the throw ability and actually using control Q produces the same results.
- You cannot put parasites inside toolboxes & other storage items

# Base Game Issues (Needs to be opened in a seperate issue, or solved here if time allows)
- non-Carrier castes should be limited to one tile egg range placing
- No doafter for placing eggs
- No sound for placing weeds
- Non-carriers should be limited to throwing eggs & weeds one tile.
- Clothing storage can be open by Xenos (excluding armour)

**Comment by Rainbeon** [comment]
>...
> You cannot put parasites inside toolboxes & other storage items

We do not need to parity obvious bugs, especially ones that don't affect the gameplay beyond accidentally griefing yourself as a new player because you put a parasite in something and lost it forever because Xenos cant take things out of inventories

**Comment by blueDev2** [comment]
@DrSmugleaf @QuietlyWhisper I think this is ready for merge. Vermida is making changes that will cover most other issues that DangerRevolution talked about that I didn't fix.

**Comment by DrSmugleaf** [comment]
Ty, I'll merge this to a different branch for verm

**Pull request merged**